test_programs/compile_success_empty/inject_context_attribute/src/main.nr

```diff
@@ -0,0 +1,55 @@
+struct Context {
+    value: Field,
+}
+
+#[inject_context]
+fn foo(x: Field) {
+    if true {
+        // 20 + 1 => 21
+        bar(qux(x + 1));
+    } else {
+        assert(false);
+    }
+}
+
+#[inject_context]
+fn bar(x: Field) {
+    let expected = _context.value;
+    assert_eq(x, expected);
+}
+
+#[inject_context]
+fn qux(x: Field) -> Field {
+    // 21 * 2 => 42
+    x * 2
+}
+
+fn inject_context(f: FunctionDefinition) {
+    // Add a `_context: Context` parameter to the function
+    let parameters = f.parameters();
+    let parameters = parameters.push_front((quote { _context }, quote { Context }.as_type()));
+    f.set_parameters(parameters);
+
+    // Create a new body where every function call has `_context` added to the list of arguments.
+    let body = f.body().map(mapping_function).quoted();
+    f.set_body(body);
```

Comment:
Hmm, maybe it'd make more sense for `set_body` to accept a `Expr` directly

Reply:
I was wondering the same thing. I'll change it. I guess it's better for it to accept `Expr` because it means you'll always have to do `as_expr().unwrap()` so you can never set something that's not an `Expr` (plus no need to parse it inside `set_body`).

Reply:
Well, almost: `Expr` can also be an `LValue`... though I can we can always turn that into an Expression, so I'll do that.